commit d4fa336fbcc388f89095b184ba6d99422cfc676c
Author: Zoltan Herczeg <hzmester@freemail.hu>
Date:   Thu Mar 24 05:34:42 2022 +0000

    Fix incorrect value reading in JIT.

commit 50a51cb7e67268e6ad417eb07c9de9bfea5cc55a
Author: Zoltan Herczeg <hzmester@freemail.hu>
Date:   Wed Mar 23 07:53:25 2022 +0000

    Fixed a unicode properrty matching issue in JIT
diff --git a/ChangeLog b/ChangeLog
index f0eab7ba..ffb08805 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -92,6 +92,9 @@ pattern, the optimizing "must be present for a match" character check was not
 being flagged as caseless, causing some matches that should have succeeded to
 fail.
 
+23. Fixed a unicode properrty matching issue in JIT. The character was not
+fully read in caseless matching.
+
 
 Version 10.39 29-October-2021
 -----------------------------
diff --git a/src/pcre2_jit_compile.c b/src/pcre2_jit_compile.c
index e7dd26c5..7fcdac86 100644
--- a/src/pcre2_jit_compile.c
+++ b/src/pcre2_jit_compile.c
@@ -7489,7 +7489,7 @@ while (*cc != XCL_END)
     {
     SLJIT_ASSERT(*cc == XCL_PROP || *cc == XCL_NOTPROP);
     cc++;
-    if (*cc == PT_CLIST)
+    if (*cc == PT_CLIST && cc[-1] == XCL_PROP)
       {
       other_cases = PRIV(ucd_caseless_sets) + cc[1];
       while (*other_cases != NOTACHAR)
diff --git a/src/pcre2_jit_test.c b/src/pcre2_jit_test.c
index 3b57ce29..8450f0b6 100644
--- a/src/pcre2_jit_test.c
+++ b/src/pcre2_jit_test.c
@@ -415,6 +415,7 @@ static struct regression_test_case regression_test_cases[] = {
 	{ MUP, 0, 0, 0 | F_NOMATCH, "[^\\p{Hangul}\\p{Z}]", " " },
 	{ MUP, 0, 0, 0, "[\\p{Lu}\\P{Latin}]+", "c\xEA\xA4\xAE,A,b" },
 	{ MUP, 0, 0, 0, "[\\x{a92e}\\p{Lu}\\P{Latin}]+", "c\xEA\xA4\xAE,A,b" },
+	{ CMUP, 0, 0, 0, "[^S]\\B", "\xe2\x80\x8a" },
 
 	/* Possible empty brackets. */
 	{ MU, A, 0, 0, "(?:|ab||bc|a)+d", "abcxabcabd" },
